contracts/USDCFlow.cdc

```diff
@@ -18,159 +20,207 @@ import "FiatToken"
-                        receiverLinkedType: Type<&USDCFlow.Vault{FungibleToken.Receiver}>(),
-                        metadataLinkedType: Type<&USDCFlow.Vault{FungibleToken.Balance, MetadataViews.Resolver}>(),
-                        providerLinkedType: Type<&USDCFlow.Vault{FungibleToken.Provider}>(),
-                        createEmptyVaultFunction: (fun (): @USDCFlow.Vault {
-                            return <-USDCFlow.createEmptyVault()
-                        })
-                    )
-            }
-            return nil
+        access(all) fun resolveView(_ view: Type): AnyStruct? {
+            return USDCFlow.resolveContractView(resourceType: nil, viewType: view)
         }
     }
 
-    /// createEmptyVault
-    ///
-    /// @return The new Vault resource with a balance of zero
-    ///
-    pub fun createEmptyVault(): @Vault {
-        return <-create Vault(balance: 0.0)
-    }
+    access(all) resource Minter: FlowEVMBridgeHandlerInterfaces.TokenMinter {
 
-    /// wrapFiatToken
-    ///
-    /// Provides a way for users to exchange a FiatToken Vault
-    /// for a USDCFlow Vault with the same balance
-    pub fun wrapFiatToken(_ from: @FungibleToken.Vault): @Vault {
-        post {
-            result.balance == before(from.balance):
-                "The USDCFlow Vault that was returned does not have the same balance as the Vault that was deposited!"
+        /// Required function for the bridge to be able to work with the Minter
+        access(all) view fun getMintedType(): Type {
+            return Type<@USDCFlow.Vault>()
         }
 
-        let vault <- from as! @FiatToken.Vault
+        /// Function for the bridge to mint tokens that are bridged from Flow EVM
+        access(FlowEVMBridgeHandlerInterfaces.Mint) fun mint(amount: UFix64): @{FungibleToken.Vault} {
+            let newTotalSupply = USDCFlow.totalSupply + amount
+            USDCFlow.totalSupply = newTotalSupply
 
-        // Get a reference to the contract account's stored Vault
-        let fiatTokenVaultRef = self.account.borrow<&FiatToken.Vault>(from: FiatToken.VaultStoragePath)
-            ?? panic("Could not borrow reference to the owner's FiatToken Vault!")
+            let vault <-create Vault(balance: amount)
 
-        let wrappedFiatTokenVault <- create Vault(balance: vault.balance)
+            emit Minted(amount: amount, mintedUUID: vault.uuid)
+            return <-vault
+        }
 
-        emit TokensMinted(amount: wrappedFiatTokenVault.balance, depositedUUID: vault.uuid, mintedUUID: wrappedFiatTokenVault.uuid)
+        /// Function for the bridge to burn tokens that are bridged back to Flow EVM
+        access(all) fun burn(vault: @{FungibleToken.Vault}) {
+            let toBurn <- vault as! @USDCFlow.Vault
+            let amount = toBurn.balance
 
-        fiatTokenVaultRef.deposit(from: <-vault)
+            assert(USDCFlow.totalSupply >= amount, message: "burning more than total supply")
```

Comment:
Should this be added to `burnCallback()`? If the check fails here I could just call `Burner.burn` passing this vault and bypass this check. If the amount is in fact >= totalSupply, we'd hit underflow there